### PR TITLE
Fix for attributes search

### PR DESCRIPTION
**Pull request by loicgasser**
https://github.com/geoadmin/mf-chsdi3/pull/1209 should be merged first

This is a definitive fix for https://github.com/geoadmin/mf-geoadmin3/issues/2009

Please review and merge

### CODE REVIEW

**Comment by loicgasser** [comment]
@procrastinatio @gjn same here. ;)

**Comment by gjn** [comment]
I don't dare to review. This is security relevant. Can't something like SQLAlchemy take care of such tasks?

File: chsdi/views/layers.py [hunk 1]
```diff
@@ -104,9 +104,9 @@ def legend(request):
     return response.body
 
 
-def _find_type(model, colname):
-    if hasattr(model, '__table__') and colname in model.__table__.c:
-        return model.__table__.c[colname].type
+def _find_type(model, colProp):
+    if hasattr(model, '__table__') and hasattr(model, colProp):
+        return model.get_column_by_name(colProp).type
```
Comment (loicgasser) [inline chsdi/views/layers.py]: This method name should be changed since it is confusing. I'll do it in a separate PR.

**Comment by loicgasser** [comment]
´(\w+\s(ilike|not ilike)\s(\'%.+%\')(?=.+%\'))+´

**Comment by loicgasser** [comment]
@procrastinatio please test and give me a feedback as this is a fix for
https://github.com/geoadmin/mf-chsdi3/issues/1179
https://github.com/geoadmin/mf-geoadmin3/issues/2009

**Comment by procrastinatio** [comment]
Instead of Regexes, I would define a simple SQL parser as

```python
from pyparsing import CaselessKeyword, CaselessLiteral, delimitedList, Each, Forward, \
         Group, Combine, Optional, Word, Regex , alphas,alphanums, nums, oneOf, \
         ZeroOrMore, quotedString, Upcase, restOfLine, ParseException

keywords = ["select", "from", "where", "and", "or"]
[select, _from, where, _and, _or] = [ CaselessKeyword(word) for word in keywords ]

E = CaselessLiteral("E")
arithSign = Word("+-",exact=1)
realNum = Regex(r"[+-]?(\d+\.\d*|\.\d+)([Ee][+-]?\d+)?")
intNum = Combine( Optional(arithSign) + Word( nums ) +
            Optional( E + Optional("+") + Word(nums) ) )

selectStmt = Forward()
table = column = Word(alphas)
columns = Group(delimitedList(column))
columnVal =   realNum | intNum | quotedString 

whereCond = (column + oneOf("= != < > >= <= like ilike", caseless=True) + columnVal)
whereExpr = Forward()
whereExpr = whereCond + ZeroOrMore((_and | _or) + whereCond)

selectStmt = Forward().setName("select statement")
selectStmt << (select +
        ('*' | columns).setResultsName("columns") +
        _from +
        table.setResultsName("table") +
        Optional(where + Group(whereExpr), '').setResultsName("where").setDebug(False)
        )
sqlComment = "--" + restOfLine
selectStmt.ignore( sqlComment )
```

Usage: 

```
$ tokens = selectStmt.parseString(sql)
$ print tokens.where
$  ['where', ['topelevationamsl', '>', '2000', 'and', 'registrationnumber', 'ilike', "'%ju%'"]]
```

**Comment by loicgasser** [comment]
Looks good, I am not sure it is more simple... It's a matter of taste I guess. It is a lot more code to do only half of the job, I think we should merge this one as it solves a lot of problems and improve it later with pyparsing or something else.

**Comment by loicgasser** [comment]
Thank you for your comments.

**Comment by cedricmoullet** [comment]
@loicgasser why haven't you consider @procrastinatio's proposal ?

**Comment by loicgasser** [comment]
I have, I opened another issue as I think this can be changed if needed.
https://github.com/geoadmin/mf-chsdi3/issues/1257
To many changes to be handled in this PR. I have not tested it and don't know if this actually does what I want it to do. To me it is just another way of parsing it.

**Pull request merged**